core/designsystem/src/main/java/com/napzak/market/designsystem/component/textfield/SearchTextField.kt

```diff
@@ -0,0 +1,113 @@
+package com.napzak.market.designsystem.component.textfield
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.foundation.text.KeyboardActions
+import androidx.compose.foundation.text.KeyboardOptions
+import androidx.compose.material3.Icon
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.vector.ImageVector
+import androidx.compose.ui.res.vectorResource
+import androidx.compose.ui.text.input.ImeAction
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.napzak.market.designsystem.R.drawable.ic_cancel_search_12
+import com.napzak.market.designsystem.R.drawable.ic_search_12
+import com.napzak.market.designsystem.theme.NapzakMarketTheme
+import com.napzak.market.util.android.noRippleClickable
+
+/**
+ * Search text field
+ *
+ * @param text: 검색 입력 값
+ * @param onTextChange: 검색 입력 값 변경
+ * @param hint: 힌트
+ * @param onSearchClick: 검색 버튼 / 키보드 검색 클릭
+ * @param onResetClick: 검색 입력 값 초기화
+ */
+@Composable
+fun SearchTextField(
+    text: String,
+    onTextChange: (String) -> Unit,
+    hint: String,
+    onResetClick: () -> Unit,
+    onSearchClick: () -> Unit,
+    modifier: Modifier = Modifier,
+) {
+    Box(
+        modifier = modifier,
+        contentAlignment = Alignment.Center,
+    ) {
+        NapzakDefaultTextField(
+            text = text,
+            onTextChange = onTextChange,
+            hint = hint,
+            textStyle = NapzakMarketTheme.typography.caption12sb,
+            hintTextStyle = NapzakMarketTheme.typography.caption12m,
+            modifier = Modifier
+                .clip(RoundedCornerShape(14.dp))
+                .background(NapzakMarketTheme.colors.gray50)
+                .padding(16.dp, 13.dp, 0.dp, 13.dp),
+            keyboardActions = KeyboardActions(
+                onSearch = { onSearchClick() },
```

Comment:
```suggestion
                onSearch = onSearchClick,
```
p2: 이렇게도 작성 가능할 것 같아요!

Reply:
keyboardActionScope가 있었군요ㅎㅎ;; 안일한 코드리뷰 죄송함닼😅